internal/formats/common/cyclonedxhelpers/format.go

```diff
@@ -57,18 +65,50 @@ func toComponent(p pkg.Package) cyclonedx.Component {
 	}
 }
 
+func lookupRelationship(ty artifact.RelationshipType) bool {
+	switch ty {
+	case artifact.OwnershipByFileOverlapRelationship:
+		return true
+	}
+	return false
+}
+
+func toDependencies(relationships []artifact.Relationship) []cyclonedx.Dependency {
+	result := make([]cyclonedx.Dependency, 0)
+	for _, r := range relationships {
+		exists := lookupRelationship(r.Type)
+
+		if !exists {
+			log.Warnf("unable to convert relationship from CycloneDX 1.3 JSON, dropping: %+v", r)
+			continue
+		}
+
+		innerDeps := []cyclonedx.Dependency{}
+		innerDeps = append(innerDeps, cyclonedx.Dependency{Ref: string(r.From.ID())})
+		result = append(result, cyclonedx.Dependency{
+			Ref:          string(r.To.ID()),
+			Dependencies: &innerDeps,
+		})
+	}
+	return result
+}
+
 func toBomDescriptorComponent(srcMetadata source.Metadata) *cyclonedx.Component {
 	switch srcMetadata.Scheme {
 	case source.ImageScheme:
+		bomRef, _ := artifact.IDByHash(srcMetadata.ImageMetadata.ID)
```

Comment:
Do we get in any trouble here ignoring this error and providing a `nil` `bomRef`?

Is it worth bailing altogether and not generating the component if a Ref cannot be derived for some reason and logging and error?

Reply:
I'll add a warning here, i saw similar function calls in another part. I can revisit this decision.

Reply:
@spiffcs i addressed this comment.